pkg/email/templates/client_requested.go

```diff
@@ -42,18 +42,41 @@ func init() {
 }
 
 func newClientRequestedData(_ context.Context, data email.NotificationTemplateData) (email.NotificationTemplateData, error) {
-	var nData ttnpb.CreateClientRequest
-	if err := ttnpb.UnmarshalAny(data.Notification().GetData(), &nData); err != nil {
+	var dd ttnpb.CreateClientEmailMessage
+	if err := ttnpb.UnmarshalAny(data.Notification().GetData(), &dd); err != nil {
 		return nil, err
 	}
 	return &ClientRequestedData{
 		NotificationTemplateData: data,
-		CreateClientRequest:      &nData,
+		CreateClientRequest:      dd.CreateClientRequest,
+		ApiKey:                   dd.ApiKey,
 	}, nil
 }
 
 // ClientRequestedData is the data for the client_requested email.
 type ClientRequestedData struct {
 	email.NotificationTemplateData
 	*ttnpb.CreateClientRequest
+	ApiKey string
+}
+
+func (crd *ClientRequestedData) SenderType() string {
+	if crd.Notification().GetSenderIds().IDString() == "" {
+		return "API key"
+	}
+	return "User"
+}
+
+func (crd *ClientRequestedData) SenderTypeMidSentence() string {
+	if crd.Notification().GetSenderIds().IDString() == "" {
+		return "API key"
+	}
+	return "user"
+}
+
+func (crd *ClientRequestedData) Sender() string {
+	if crd.Notification().GetSenderIds().IDString() == "" {
+		return crd.ApiKey
```

Comment:
I recommend `GetApiKey`.

Reply:
Instead of propagating the `CreateClientEmailMessage` I just added the ApiKey field on the `ClientRequestedData`, therefore there is no proto methods generated for that.

```go
// ClientRequestedData is the data for the client_requested email.
type ClientRequestedData struct {
	email.NotificationTemplateData
	*ttnpb.CreateClientRequest
	ApiKey string
}
```

Reply:
In that spirit I would keep the whole `*ttnpb.APIKey` then.